<?xml version="1.0"?>

<launch>
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="control_mode" default="twist" />

  <node name="joy" pkg="joy" type="joy_node" output="screen" >
    <param name="dev" value="$(arg joy_dev)" />
  </node>

  <!-- Note that axis IDs are those from the joystick message plus one, to be able to invert axes by specifiying either positive or negative axis numbers.-->
  <!-- Axis 2 from joy message thus has to be set as '3' or '-3'(inverted mode) below-->
  <node name="quadrotor_teleop" pkg="hector_quadrotor_teleop" type="quadrotor_teleop">
    <param name="control_mode" value="$(arg control_mode)" />
    <param name="x_axis" value="4"/>
    <param name="y_axis" value="3"/>
    <param name="z_axis" value="2"/>
    <param name="yaw_axis" value="1"/>
    <param name="x_velocity_max" value="5"/>
    <param name="y_velocity_max" value="5"/>
    <param name="z_velocity_max" value="2"/>
    <param name="slow_button" value="6"/>
  </node>
</launch>

